rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMStatsView.java

```diff
@@ -25,81 +25,93 @@
-import com.google.gwt.visualization.client.LegendPosition;
-import com.google.gwt.visualization.client.visualizations.corechart.AreaChart;
-import com.google.gwt.visualization.client.visualizations.corechart.ColumnChart;
-import com.google.gwt.visualization.client.visualizations.corechart.HorizontalAxisOptions;
-import com.google.gwt.visualization.client.visualizations.corechart.Options;
+import com.smartgwt.client.types.Alignment;
 import com.smartgwt.client.types.Overflow;
 import com.smartgwt.client.widgets.Canvas;
 import com.smartgwt.client.widgets.Label;
 import com.smartgwt.client.widgets.form.DynamicForm;
-import com.smartgwt.client.widgets.form.fields.CheckboxItem;
 import com.smartgwt.client.widgets.form.fields.SelectItem;
-import com.smartgwt.client.widgets.form.fields.events.ChangedHandler;
 import com.smartgwt.client.widgets.layout.HLayout;
 import com.smartgwt.client.widgets.layout.VLayout;
 
 
 /**
  * Displays monitoring graphs using Google Visualization
- * 
- * 
- * @author mschnoor
- *
  */
 public class RMStatsView implements StatsListener, NodesListener {
 
-    private RMController controller;
+    private static final String FREE_NODES_COUNT = "FreeNodesCount";
 
-    private AreaChart nodeLineChart;
+    private static final String NEEDED_NODES_COUNT = "NeededNodesCount";
 
-    private DataTable nodeLineTable;
+    private static final String BUSY_NODES_COUNT = "BusyNodesCount";
 
-    private Options nodeLineOpts;
+    private static final String AVAILABLE_NODES_COUNT = "AvailableNodesCount";
 
-    private int nodeLineTimeId, nodeLineFreeId, nodeLineBusyId, nodeLineDownId, nodeLineTotalId, linePendingTasksId,
-            nodeLineDeployingId, nodeLineConfiguringId, nodeLineLostId;
+    private static final String DEPLOYING_NODES_COUNT = "DeployingNodesCount";
 
-    private DynamicForm nodeLineForm;
+    private static final String CONFIG_NODES_COUNT = "ConfigNodesCount";
 
-    private Label nodeLineHeaderLabel;
+    private static final String DOWN_NODES_COUNT = "DownNodesCount";
 
-    private DynamicForm nodeLineSeriesForm;
+    private static final String LOST_NODES_COUNT = "LostNodesCount";
 
-    private ColumnChart nodeColChart;
+    private static final String AVERAGE_ACTIVITY = "AverageActivity";
 
-    private DataTable nodeColTable;
+    private RMController controller;
 
-    private Options nodeColOpts;
+    private Label nodeLineHeaderLabel;
 
-    private Label nodeColHeaderLabel;
+    private DynamicForm loadForm;
 
-    private AreaChart loadChart;
+    private Label loadHeaderLabel;
 
-    private DataTable loadTable;
+    private BarChart nodeStateChart;
 
-    private Options loadOpts;
+    private LineChart nodeHistoryChart;
 
-    private int loadTimeId, loadValId;
+    private LineChart activityChart;
 
-    private DynamicForm loadForm;
+    private DynamicForm nodeLineForm;
 
-    private Label loadHeaderLabel;
+    private List<Map.Entry<String, String>> datasetsAndColor = new ArrayList<>();
```

Comment:
You should put the size of the ArrayList

Reply:
why tho?

Reply:
If you don't set the size of the array the JVM may have to reallocate space each time you add an item. When the size is known it is better to set the size to avoid such reallocation.

Reply:
ok i see you are really concern with ArrayList initialization